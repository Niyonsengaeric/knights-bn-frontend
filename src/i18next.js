import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import Backend from 'i18next-http-backend';
import LanguageDetector from 'i18next-browser-languagedetector';

i18n
  .use(Backend)
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    fallbackLng: 'en',
    backend: {
      loadPath: '../public/locales/{{lng}}/{{ns}}.json',
    },
    react: {
      useSuspense: false,
    },
    debug: false,
    interpolation: {
      escapeValue: false,
    },
  });


export default i18n;
