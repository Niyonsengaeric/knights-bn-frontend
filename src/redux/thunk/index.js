import axios from 'axios';
import localStorage from 'localStorage';
import decodeJwt from 'jwt-decode';
import { setCurentUser, clearErrorMsg } from '../actions/actions';

const userToken = localStorage.getItem('user-token');
axios.defaults.headers['user-token'] = userToken;

const axiosInstance = axios.create({
  baseURL: 'https://knights-bn-backnd.herokuapp.com/api/v1',
});

const authThunk = (
  method,
  path,
  action,
  body,
) => (dispatch) => {
  dispatch(action({ loading: true }));
  return axiosInstance[method](path, body)
    .then((response) => {
      if (!response.data.token) dispatch(action({ ...response.data, loading: false }));
      localStorage.setItem('user-token', response.data.token);
      dispatch(action({
        ...decodeJwt(response.data.token),
        ...response.data,
        isAuthenticated: true,
        loading: false,
      }));
    })
    .catch((error) => {
      dispatch(action({
        ...error.response.data,
        isAuthenticated: false,
        loading: false,
      }));
    });
};

const callApiThunk = (method, path, action, body) => (dispatch) => axiosInstance[method](path, body)
  .then((response) => {
    dispatch(action({ ...response.data, isLoggedIn: true, loading: false }));
  })
  .catch((error) => {
    dispatch(action({ ...error.response.data, isLoggedIn: false, loading: false }));
  });

const resetPassword = (
  method,
  path,
  action,
  body,
  token,
) => (dispatch) => {
  dispatch(action({ loading: true }));
  return axiosInstance[method](path, body).then((response) => {
    localStorage.setItem('user-token', token);
    dispatch(action({
      ...decodeJwt(token), ...response.data, isAuthenticated: true, loading: false,
    }));
  }).catch((error) => {
    dispatch(action({
      ...error.response.data, isAuthenticated: false, loading: false,
    }));
  });
};

const logout = () => (dispatch) => {
  localStorage.removeItem('user-token');
  dispatch(setCurentUser({ isAuthenticated: false }));
};
const clearMessage = () => (dispatch) => {
  dispatch(clearErrorMsg());
};
export {
  axiosInstance,
  authThunk,
  callApiThunk,
  resetPassword,
  logout,
  clearMessage,
};
