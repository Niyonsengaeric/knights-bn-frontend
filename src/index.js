import React, { Suspense } from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import localStorage from 'localStorage';
import jwtDecode from 'jwt-decode';
import App from './route/App';
import configureStore from './redux';
import './i18next';
import { setCurentUser } from './redux/actions/actions';

const store = configureStore();
if (localStorage.getItem('user-token')) {
  store.dispatch(setCurentUser({ ...jwtDecode(localStorage.getItem('user-token')), isAuthenticated: true }));
}
ReactDOM.render(
  <Suspense fallback="Loading ...">
    <Provider store={store}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>
  </Suspense>,
  document.getElementById('root'),
);
