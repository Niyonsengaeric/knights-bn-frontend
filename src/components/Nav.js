import React from 'react';
import '../assets/styles/components/navigationlinks.scss';
import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import { connect } from 'react-redux';
import Languages from './Languages';
import logo from '../assets/images/logo@2x.png';
import { logout } from '../redux/thunk/index';

const Nav = ({ isLoggedIn, logout, role }) => {
  const { t } = useTranslation();
  const notAuth = (
    <ul>
      <li><Link to="/login" className="nav-link">{t('Login.1')}</Link></li>
      <li><Link to="/signup" className="nav-link">{t('Sign up.1')}</Link></li>
    </ul>
  );
  const isAuth = (
    <ul>
      <li><Link to="/" id="logout-link" onClick={logout} className="nav-link">logout</Link></li>
      <li><Link to="/admin" id="users-link" className="nav-link">{(role === 'superAdmin') ? 'AllUsers' : ''}</Link></li>
      <li><Link to="/profile" id="logout-link" className="nav-link">Profile</Link></li>
      <Languages />
    </ul>
  );
  return (
    <div className="main-navigation">
      <div className="nav-menu">
        <div className="web-logo">
          <Link to="/" className="nav-link-img">
            <img className="logo-img" src={logo} />
          </Link>
        </div>
        <div className="right-menu">
          <Languages />
          {isLoggedIn ? isAuth : notAuth}

        </div>

      </div>
    </div>
  );
};

const mapStateToProps = (state) => ({
  isLoggedIn: state.AuthReducer.isAuthenticated,
  role: state.AuthReducer.role,
});

export default connect(mapStateToProps, { logout })(Nav);
