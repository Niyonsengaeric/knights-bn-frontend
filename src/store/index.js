import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import reducers from './reducers/index';

const initialState = {};

const store = createStore(
  reducers,
  initialState,
  composeWithDevTools(applyMiddleware(thunk)),
);

export default store;
