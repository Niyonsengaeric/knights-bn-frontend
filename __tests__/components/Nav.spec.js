
import { Provider } from 'react-redux';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import '../../src/i18next';
import renderer from 'react-test-renderer';
import Nav from '../../src/components/Nav';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);


describe('Test Nav component', () => {
  it('should render Nav component', () => {
    const store = mockStore(
      {
        AuthReducer: { isAuthenticated: false },
      },
    );
    const NavComponent = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(NavComponent).toMatchSnapshot();
  });
  it('should render Nav component', () => {
    const store = mockStore(
      {
        AuthReducer: {
          isAuthenticated: true,
          role: 'superAdmin',
        },
      },
    );
    const NavComponent = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(NavComponent).toMatchSnapshot();
  });
});
