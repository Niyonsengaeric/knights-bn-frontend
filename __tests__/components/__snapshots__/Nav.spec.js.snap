// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Nav component should render Nav component 1`] = `
<div
  className="main-navigation"
>
  <div
    className="nav-menu"
  >
    <div
      className="web-logo"
    >
      <a
        className="nav-link-img"
        href="/"
        onClick={[Function]}
      >
        <img
          className="logo-img"
          src=""
        />
      </a>
    </div>
    <div
      className="right-menu"
    >
      <select
        className="lang lang--list"
        onChange={[Function]}
      >
        <option>
          Choose language.1
        </option>
        <option
          value="ki"
        >
          ikinyarwanda 
        </option>
        <option
          value="en"
        >
          english
        </option>
        <option
          value="fr"
        >
          français
        </option>
      </select>
      <ul>
        <li>
          <a
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login.1
          </a>
        </li>
        <li>
          <a
            className="nav-link"
            href="/signup"
            onClick={[Function]}
          >
            Sign up.1
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Test Nav component should render Nav component 2`] = `
<div
  className="main-navigation"
>
  <div
    className="nav-menu"
  >
    <div
      className="web-logo"
    >
      <a
        className="nav-link-img"
        href="/"
        onClick={[Function]}
      >
        <img
          className="logo-img"
          src=""
        />
      </a>
    </div>
    <div
      className="right-menu"
    >
      <select
        className="lang lang--list"
        onChange={[Function]}
      >
        <option>
          Choose language.1
        </option>
        <option
          value="ki"
        >
          ikinyarwanda 
        </option>
        <option
          value="en"
        >
          english
        </option>
        <option
          value="fr"
        >
          français
        </option>
      </select>
      <ul>
        <li>
          <a
            className="nav-link"
            href="/"
            id="logout-link"
            onClick={[Function]}
          >
            logout
          </a>
        </li>
        <li>
          <a
            className="nav-link"
            href="/admin"
            id="users-link"
            onClick={[Function]}
          >
            AllUsers
          </a>
        </li>
        <li>
          <a
            className="nav-link"
            href="/profile"
            id="logout-link"
            onClick={[Function]}
          >
            Profile
          </a>
        </li>
        <select
          className="lang lang--list"
          onChange={[Function]}
        >
          <option>
            Choose language.1
          </option>
          <option
            value="ki"
          >
            ikinyarwanda 
          </option>
          <option
            value="en"
          >
            english
          </option>
          <option
            value="fr"
          >
            français
          </option>
        </select>
      </ul>
    </div>
  </div>
</div>
`;
